### PR TITLE
Implement PKCS

### PR DESCRIPTION
**Pull request by adidiner**

### CODE REVIEW

File: hw1/PKCS-1.5.py [hunk 1]
```diff
@@ -121,8 +123,10 @@ def enc_PKCS_1(self, d, ps=None):
             raise Exception("first byte must be nonzero 0 if bt=0")
 
         if ps is None:
-            ps = ?
-        eb = ?  # Encryption Block
+            # PS is at least `RSA_PKCS_1.min_pad_size-3` long, as
+            # it was validated that d is short enough.
+            ps = self.pad(self.k - 3 - len(d))
+        eb = b'\x00' + self.bt.to_bytes(1, byteorder='big') + ps + b'\x00' + d
```
Comment (szaccount) [inline hw1/PKCS-1.5.py:133]: Add comment that if ps is passed it is not validated as it was implied from the requirements to only check the decryption and it is also usefull for testing to not check it.

Comment (adidiner) [reply hw1/PKCS-1.5.py:133]: Done

File: hw1/PKCS-1.5.py [hunk 2]
```diff
@@ -176,7 +180,70 @@ def parse(self, eb):
...
+        # Validate that the block is of valid size.
+        if len(eb) != self.k:
+            return None
+        # Validate that the first byte is 00.
+        first_byte = eb[0]
+        if first_byte != 0:
+            return None
+        # Validate matching BT.
+        # The BT should be only 2 for private-key decryption,
+        # 0 and 1 are supported for completness (for public-key decryption).
```
Comment (szaccount) [inline hw1/PKCS-1.5.py]:
```suggestion
        # 0 and 1 are supported for completness (for public-key decryption, for signatures).
```

Comment (adidiner) [reply hw1/PKCS-1.5.py]: Done

**Pull request merged**